vividus-plugin-rest-api/src/main/java/org/vividus/http/MimeTypeUtils.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HeaderElement;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.message.MessageSupport;
+
+public final class MimeTypeUtils
+{
+    private MimeTypeUtils()
+    {
+    }
+
+    public static Optional<String> getMimeTypeFromHeaders(Header... headers)
+    {
+        return Stream.of(headers)
+                .filter(h -> HttpHeaders.CONTENT_TYPE.equalsIgnoreCase(h.getName())
+                        && StringUtils.isNotBlank(h.getValue()))
+                .findFirst()
+                .map(MessageSupport::parse)
+                .map(elements -> elements[0])
+                .map(HeaderElement::getName);
```

Comment:
```suggestion
                .map(HeaderElement::getName)
                .orElseGet(ContentType.DEFAULT_TEXT::getMimeType);
```
and then add javadocs describing the current behavior

Reply:
> and then add javadocs describing the current behavior

We have one place where we get type with additional logic
https://github.com/vividus-framework/vividus/pull/4537/files#diff-dd131ecfa6f29400ddeb501304099af8e36c2942b87e6b5f47490af5ee1bc919L68
So I decided to create separate utils method